tools/ci/check_grep2.py

```diff
@@ -0,0 +1,519 @@
+from pathlib import Path
+import re
+import sys
+import time
+from collections import OrderedDict
+
+RED = "\033[0;31m"
+GREEN = "\033[0;32m"
+BLUE = "\033[0;34m"
+NC = "\033[0m"  # No Color
+
+MAP_COMMENT = "//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE"
+IGNORE_515_PROC_MARKER = "__byond_version_compat.dm"
+
+#
+# Basic map issues
+#
+# Basic map issues operate on a line-by-line basis.
+#
+def check_tgm(filename, lines):
+    for line in lines:
+        if re.match(r"^\".+\" = \(.+\)", line):
+            return "ERROR: Non-TGM formatted map detected. Please convert it using Map Merger!"
+
+
+def check_comments(filename, lines):
+    for line in lines:
+        if (
+            re.match(r"//", line)
+            and not re.match(MAP_COMMENT, line)
+            and not re.match("name|desc", line)
+        ):
+            return "ERROR: Unexpected commented out line detected in this map file. Please remove it."
+
+
+def check_conflict_markers(filename, lines):
+    for line in lines:
+        if "Map Merge Marker" in line:
+            return "ERROR: Merge conflict markers detected in map, please resolve all merge failures!"
+
+
+def check_conflict_marker_object(filename, lines):
+    for line in lines:
+        if "/obj/merge_conflict/marker" in line:
+            return "ERROR: Merge conflict markers detected in map, please resolve all merge failures!"
+
+
+def check_iconstate_tags(filename, lines):
+    for line in lines:
+        if re.match(r'^\ttag = "icon', line):
+            return "ERROR: Tag vars from icon state generation detected in maps, please remove them."
+
+
+def check_step_varedits(filename, lines):
+    for line in lines:
+        if re.match(r"step_[xy]", line):
+            return (
+                "ERROR: step_x/step_y variables detected in maps, please remove them."
+            )
```

Comment:
```suggestion
            return "ERROR: step_x/step_y variables detected in maps, please remove them."
```
is there a specific reasons why this is formatted differently than the other return statements?

Reply:
I was using black for formatting but I don't like how it handles these. I won't run a preformatter for now and maybe I can land on a coding style for this and the other Python code in the repo.